source/client/process_impl.cc

```diff
@@ -338,6 +338,71 @@ getUserDefinedFactoryConfigPairs(const Options& options) {
   return factory_config_pairs;
 }
 
+/**
+ * Takes in a single worker's user defined outputs and collects them into the
+ * user_defined_outputs_by_plugin map. When called for each worker, transforms the outputs from
+ * being mapped by Worker to being mapped by Plugin.
+ *
+ * @param user_defined_outputs_by_plugin The map that this function will collect the worker data
+ * into, maps plugin name to the set of results for that plugin.
+ * @param worker_user_defined_outputs The per worker results to collect and organize.
+ */
+void collectUserDefinedOutputs(
+    absl::flat_hash_map<std::string, std::vector<nighthawk::client::UserDefinedOutput>>*
+        user_defined_outputs_by_plugin,
+    const std::vector<nighthawk::client::UserDefinedOutput>& worker_user_defined_outputs) {
+  for (const nighthawk::client::UserDefinedOutput& user_defined_output :
+       worker_user_defined_outputs) {
+    std::vector<nighthawk::client::UserDefinedOutput> cross_worker_results_for_plugin =
```

Comment:
This is a bit long for me to easily parse the ternary expression. Would you mind expanding it into an if<->else? It will take more space horizontally, but should be easier to follow.

(optional) Consider taking the portion that finds a previous output by plugin name into its own private function. That would improve the readability here further.